samples/apps/copilot-chat-app/webapi/Controllers/BotController.cs

```diff
@@ -0,0 +1,254 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Memory;
+using SemanticKernel.Service.Config;
+using SemanticKernel.Service.Model;
+using SemanticKernel.Service.Skills;
+using SemanticKernel.Service.Storage;
+
+namespace SemanticKernel.Service.Controllers;
+
+[ApiController]
+public class BotController : ControllerBase
+{
+    private readonly IConfiguration _configuration;
+    private readonly ILogger<SemanticKernelController> _logger;
+    private readonly IMemoryStore _memoryStore;
+    private readonly ChatSessionRepository _chatRepository;
+    private readonly ChatMessageRepository _chatMessageRepository;
+
+    /// <summary>
+    /// The constructor of BotController.
+    /// </summary>
+    /// <param name="configuration">The application configuration.</param>
+    /// <param name="memoryStore">The memory store.</param>
+    /// <param name="chatRepository">The chat session repository.</param>
+    /// <param name="chatMessageRepository">The chat message repository.</param>
+    /// <param name="logger">The logger.</param>
+    public BotController(IConfiguration configuration, IMemoryStore memoryStore, ChatSessionRepository chatRepository, ChatMessageRepository chatMessageRepository, ILogger<SemanticKernelController> logger)
+    {
+        this._configuration = configuration;
+        this._logger = logger;
+        this._memoryStore = memoryStore;
+        this._chatRepository = chatRepository;
+        this._chatMessageRepository = chatMessageRepository;
+    }
+
+    /// <summary>
+    /// Upload a bot.
+    /// </summary>
+    /// <param name="kernel">The Semantic Kernel instance.</param>
+    /// <param name="userId">The user id.</param>
+    /// <param name="bot">The bot object from the message body</param>
+    /// <returns>The HTTP action result.</returns>
+    [Authorize]
+    [HttpPost]
+    [Route("bot/upload")]
+    [ProducesResponseType(StatusCodes.Status202Accepted)]
+    [ProducesResponseType(StatusCodes.Status415UnsupportedMediaType)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult> UploadAsync(
+        [FromServices] Kernel kernel,
+        [FromQuery] string userId,
+        [FromBody] Bot bot)
+    {
+        // TODO: We should get userId from server context instead of from request for privacy/security reasons when support multipe users.
```

Comment:
Is this relevant for this API call?

Reply:
Yes, it's for the userId that we currently get from the request query string. I've log a task to address those as a refactory.